core/app/models/spree/store_credit_category.rb

```diff
@@ -4,12 +4,23 @@ class Spree::StoreCreditCategory < Spree::Base
   class_attribute :non_expiring_credit_types
   self.non_expiring_credit_types = [Spree::StoreCreditType::NON_EXPIRING]
 
+  # @deprecated
+  # Please use Spree::Reimbursement.store_credit_category_proc= instead.
   class_attribute :reimbursement_category_name
-  self.reimbursement_category_name = I18n.t('spree.store_credit_category.default')
 
-  def self.reimbursement_category(_reimbursement)
-    Spree::StoreCreditCategory.find_by(name: reimbursement_category_name) ||
-      Spree::StoreCreditCategory.first
+  # @deprecated
+  # Please use Spree::Reimbursement#store_credit_category instead.
+  def self.reimbursement_category(reimbursement)
+    reimbursement.store_credit_category
+  end
+
+  deprecate :reimbursement_category_name, deprecator: Spree::Deprecation
+  deprecate :reimbursement_category_name=, deprecator: Spree::Deprecation
+  deprecate :reimbursement_category_name?, deprecator: Spree::Deprecation
+  class << self
+    deprecate :reimbursement_category_name, deprecator: Spree::Deprecation
+    deprecate :reimbursement_category_name=, deprecator: Spree::Deprecation
+    deprecate :reimbursement_category, deprecator: Spree::Deprecation
```

Comment:
Can you add a message explaining what should be used instead? Otherwise, users may think these have been deprecated with no replacement, unless they read the documentation.